Python/Product/Analysis/Pythia/ExpressionWalker.cs

```diff
@@ -0,0 +1,167 @@
+﻿// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.PythonTools.Parsing.Ast;
+
+namespace Microsoft.PythonTools.Analysis.Pythia {
+    /// <summary>
+    /// AST Expression walker for expression assignments
+    /// </summary>
+    sealed class ExpressionWalker : PythonWalker {
+        private IList<KeyValuePair> Assignments;
+        private List<KeyValuePair> MethodInvocations;
+
+        private IList<Tuple<int, int>> ConditionalRanges;
```

Comment:
Can you define a type instead of anonymous tuple?

Reply:
Pythia's folks code so I am not inclined to massage it too much :-)